lib/AST/ASTStructuralEquivalence.cpp

```diff
@@ -831,6 +831,67 @@ static bool IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
   return true;
 }
 
+/// Determine structural equivalence of two methodss.
+static bool IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
+                             CXXMethodDecl *Method1, CXXMethodDecl *Method2) {
+  if (Method1->isStatic() != Method2->isStatic())
```

Comment:
Could we check for `&&` qualifier as well?
Is `noexcept` checked during normal function type check (in the call of line 888)?